js/product-details.js

```diff
@@ -0,0 +1,255 @@
+const product = location.search.substr(1).split("_");
+const productCategory = product[0];
+const productId = product[1];
+var drink = "";
+
+var xmlhttp = new XMLHttpRequest();
+xmlhttp.onreadystatechange = function() {
+  if (this.readyState == 4 && this.status == 200) {
+    var data = JSON.parse(this.responseText);
+    drink = data["drinks"][productId].additionalDescription;
+    for (key in drink) {
+      if (drink[key] == "") {
+      } else {
+        bildDetailButton();
+        bildDetailvalueConteiners();
+      }
+    }
+    clickButton();
+    if (windowWidth() > 769) {
+      let a = document.getElementsByClassName("details-description")[0];
+      a.click();
+    }
+  }
+};
+xmlhttp.open("GET", "jsons/" + productCategory + ".json", true);
+xmlhttp.send();
+
+const bildDetailButton = function() {
+  var productDetails = document.getElementsByClassName("product-details")[0];
+  let detailButton = document.createElement("div");
+  detailButton.innerHTML = key;
+  detailButton.classList.add("details-" + key);
+  detailButton.classList.add("details-button");
+  productDetails.appendChild(detailButton);
+};
+
+const bildDetailvalueConteiners = function() {
+  const conteinerSmoll = document.getElementsByClassName("product-details")[0];
+  const conteinerBig = document.getElementsByClassName(
+    "product-details-value"
+  )[0];
+  if (Object.prototype.toString.call(drink[key]) == "[object Object]") {
+    let drinkKey = drink[key];
+    let valueConteinerSmoll = document.createElement("div");
+    valueConteinerSmoll.classList.add("product-details-value--smoll");
+    valueConteinerSmoll.classList.add("product-details-value--smoll--none");
+    valueConteinerSmoll.classList.add("product-details-value--smoll--" + key);
+    let valueConteinerBig = document.createElement("div");
+    valueConteinerBig.classList.add("product-details-value--big");
+    valueConteinerBig.classList.add("product-details-value--big--none");
+    valueConteinerBig.classList.add("product-details-value--big--" + key);
+    for (key in drinkKey) {
```

Comment:
when you're using for in you need to check if current object has this property in his own
if (drinkKey.hasOwnProperty(key))
it's important